UI-Update/Query-API-Modification

Karthi-DStech opened the pull request:
Greetings, 

I have implemented the following features and cleared bugs in the Lettuce Codebase which was mentioned below: 

- Imformal Names are Queried first using OMOP Database. 

- Made changes in the User Interface that user's will get results directly from the OMOP if they entered the formal name which matches correctly with the database. 

- Updated GUI so that all the results are shown to the human in the loop as well. 

- If no output was returned from the user's input, an option was provided where the user can select whether to use LLM or Not. If the user choose "yes", the LLM is used to get the formal name and then Queried first for OMOP Database. 

- Optimised the API when fetching results for multiple informal names (as API was called every time for each informal names. For eg: if the user provied 3 informal names, the API called for 3 times and results were fetched). 

  As a solution, I made the API to take the informal names as a list and called once which results in avoiding the API calls for multiple times. 

- I have also added logics if the user selects "No", then they will given an option to end the session and it returns a thank you messgae. 

I have also tested the code multiple times and it works very well. Kindly review the code and do the needful.

kuraisle commented on Carrot-Assistant/app.py:
```diff
@@ -136,16 +136,22 @@ def parse_pipeline_args(base_options: BaseOptions, options: PipelineOptions) ->
-    Parameters
+    Generate LLM output and OMOP results for a list of informal names.
+    
+    parameters
```
Capitalise, please

kuraisle commented on Carrot-Assistant/app.py:
```diff
@@ -191,52 +199,78 @@ async def generate_events(request: PipelineRequest) -> AsyncGenerator[str]:
 
 @app.post("/run")
-async def run_pipeline(request: PipelineRequest) -> EventSourceResponse:
+async def run_pipeline(request: Request) -> EventSourceResponse:
```
Can you explain this change? Why doesn't the pipeline use a PipelineRequest?

Karthi-DStech replied:
I believe before modification, we couldn't modify the API as the LLM uses the medication name to predict the formal name and queries the OMOP Database. By using "Request", I can send it to the OMOP Querying and if necessary (to use LLM), the "Request" will be convert into "PipelineRequest".

kuraisle commented:
> As you can see, we have implemented the logic of fixing multiple queries in this commit.

If you could review #41 , you'll see a refactor. If it passes review, we can merge it and work from there. Also, if you recall, we discussed the fact we can't rely on users having database credentials, so calling omop directly from the UI isn't how we want to do this

Karthi-DStech commented:
In this commit, I've solved multiple bugs and added new features. Most of my logic still relied on app.py. In this commit, I've just added some new variables to fetch the results in GUI. 

But as we can see,  you've moved everything to the router. Can I implement the same solution in the router's logic because I've not made any changes or used (OMOP_Matcher) to control the logic as I did earlier in #30.

kuraisle commented:
OK so I've had a look.

The run endpoint now has the logic:
1. Query OMOP
2. If use_llm then try the LLM, then query OMOP for the LLM's guess

When I tried it, when I sent the use_llm request, the whole thing ran, so I ended up querying OMOP twice with the input.

Why not just make a request to run_db, then collect all the ones that want LLM queries in the UI, then make another request to the full pipeline?

Karthi-DStech commented:
Hi James, 

I have updated the scripts and it works fine now. You can check once and push it on the main.
